lib/driver.js

```diff
@@ -244,6 +276,11 @@ class EspressoDriver extends BaseDriver {
       }
     };
 
+    const unzippedAppPath = await this.unzipAppIfNeeded(appPath);
```

Comment:
Please merge the unzipping logging with the caching logic below. Basically we must not unzip an app again if it was already cached and the app must be properly presigned.